wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift

```diff
@@ -1137,7 +1137,9 @@ extension ZMUserSession: SyncAgentDelegate {
         syncContext.performGroupedBlock { [weak self] in
             guard let self else { return }
             WireLogger.sync.debug("did finish incremental sync")
-            processLegacyEvents()
+            if !journal[.isSyncV2Enabled] {
+                processLegacyEvents()
+            }
```

Comment:
todo: it should be done within processLegacyEvents() as it's called from multiple place